src/App.js

```diff
@@ -41,24 +40,27 @@ function App() {
   }, []);
 
   const handleSearch = (searchValue) => {
-    const lowercaseSearch = searchValue.toLowerCase();
+  const searchTerms = searchValue.toLowerCase().split(" ");
     const results = [];
-
     for (const object of combinedData) {
       const lowercaseName = object.name.toLowerCase();
       const lowercaseLocation = object.location.toLowerCase();
-      const matchingSkills = object.skills.filter((skill) =>
-        skill.toLowerCase().includes(lowercaseSearch)
-      );
-      if (
-        matchingSkills.length > 0 ||
-        lowercaseName.includes(lowercaseSearch) ||
-        lowercaseLocation.includes(lowercaseSearch)
-      ) {
-        results.push(object);
+      // For each search term, check if it matches any skills, name, or location
+      for (const term of searchTerms) {
+        const matchingSkills = object.skills.filter((skill) =>
+          skill.toLowerCase().includes(term)
```

Comment:
```suggestion
        const matchingSkills = object.skills.filter((skill) =>
          skill.toLowerCase().includes(term)
```

Reply:
I think we can move this line outside the `term` loop. So it will run only time rather than running for every character.